crates/astria-conductor/src/sequencer/mod.rs

```
@@ -0,0 +1,252 @@
+                    if let Err(e) = forward_pending_block(executor_tx.clone(), res) {
+                        error!("failed forwarding blocks during regular operation; exiting reader");
+                        break 'reader_loop Err(e).wrap_err("failed forwarding blocks regular operation");
+                    }
+                }
+
+                res = &mut resubscribe => {
+                    if let Err(e) = assign_new_blocks_subscription(&mut new_blocks, res) {
+                        error!(error.message = %e, error.cause = ?e, "failed to resubscribe to get new blocks from sequencer; exiting reader");
+                        break 'reader_loop Err(e).wrap_err("failed to resubscribe to new blocks from sequencer");
+                    }
+                }
+            }
+        }
+    }
+}
+
+fn assign_new_blocks_subscription(
+    subscription: &mut stream::Fuse<NewBlocksStream>,
+    res: eyre::Result<NewBlocksStream>,
+) -> eyre::Result<()> {
+    use futures::stream::StreamExt as _;
+
+    let new_subscription = res.wrap_err("failed subscribing to new blocks from sequencer")?;
+    *subscription = new_subscription.fuse();
+
+    Ok(())
+}
+
+async fn subscribe_new_blocks(pool: Pool<ClientProvider>) -> eyre::Result<NewBlocksStream> {
+    use sequencer_client::SequencerSubscriptionClientExt as _;
+    pool.get()
+        .await
+        .wrap_err("failed getting a sequencer client from the pool")?
+        .subscribe_new_block_data()
+        .await
+        .wrap_err("failed subscribing to sequencer to receive new blocks")
+}
+
+fn forward_pending_block(
+    executor_tx: executor::Sender,
+    res: Result<SequencerBlockData, NewBlockStreamError>,
+) -> eyre::Result<()> {
+    let block = match res {
+        Err(e) => {
+            warn!(error.message = %e, error.cause = ?e, "response from sequencer block subscription was bad; dropping it");
+            return Ok(());
```

shouldn't this case try to resubscribe instead of silently dropping?

---

This case means that a) we got an actual response from the new block stream, but that b) the data we received was bad.

So the stream is still active, but the returned data is bad and we have to drop it.